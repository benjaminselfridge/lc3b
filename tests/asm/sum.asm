        0x3000

START:  LEA     R0      INPUT
        LDW     R0      R0      0x0 ; R0 will contain the factorial

        ADD     R1      R0      0x0 ; R1 will be decremented at each iteration

LOOP:   ADD     R1      R1      -1
        BRnz    DONE
        ADD     R0      R1      R0
        BR      LOOP

DONE:   TRAP 0x0
        
        
INPUT:  8                     ; compute 8 + 7 + .. + 1
