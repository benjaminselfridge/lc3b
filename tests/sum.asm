        0x3000
START:  AND  R0 R0 0x0          ; R0 <- 0x0
        ADD  R0 R0 0x10         ; R0 <- 0x10
        AND  R1 R1 0x0          ; R1 <- 0 (sum)
LOOP:   AND  R0 R0 R0           ; Set NZP on R0
        BRnz END                ; If R0 <= 0, jump to end
        ADD  R1 R1 R0           ; R1 += R0
        ADD  R0 R0 0xFFFF       ; R0 -= 1
        JMP LOOP
END:    TRAP 0x0
