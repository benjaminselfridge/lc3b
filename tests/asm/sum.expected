[ RegContains 0 36 ]